product_variant_sale_price/models/product_product.py

```diff
@@ -0,0 +1,61 @@
+# -*- coding: utf-8 -*-
+# © 2016 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from openerp import models, fields, api
+import openerp.addons.decimal_precision as dp
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    @api.multi
+    def write(self, vals):
+        res = super(ProductTemplate, self).write(vals)
+        if 'list_price' in vals:
+            for variant in self.mapped('product_variant_ids'):
+                variant._onchange_impact_price()
+                variant._onchange_lst_price()
+        return res
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.multi
+    @api.depends('fix_price')
+    def _compute_lst_price(self):
+        for product in self:
+            price = product.fix_price or product.list_price
+            if 'uom' in self.env.context:
+                uom = product.uos_id or product.uom_id
+                price = uom._compute_price(price, self.env.context['uom'])
```

Comment:
I think here you missed ```product.uom_id.id``` as first argument. Please also compare to https://github.com/sergio-teruel/product-variant/blob/8.0-PR-product_variant_sale_price/product_variant_sale_price/models/product_product.py#L32